src/Psalm/Internal/Scanner/DocblockParser.php

```diff
@@ -87,15 +88,15 @@ public static function parse(string $docblock, int $offsetStart): ParsedDocblock
                 }
             }
 
-            if (preg_match('/^[ \t]*\*?\s*@([\w\-\\\:]+)[\t ]*(.*)$/sm', $line, $matches, PREG_OFFSET_CAPTURE)) {
+            if (preg_match('/^[ \t]*\*?\s*@([\w\-\\\:]+)[\t ]*(.*)$/', $line, $matches, PREG_OFFSET_CAPTURE)) {
                 /** @var array<int, array{string, int}> $matches */
                 [, $type_info, $data_info] = $matches;
 
                 [$type] = $type_info;
                 [$data, $data_offset] = $data_info;
 
-                if (strpos($data, '*')) {
-                    $data = rtrim(preg_replace('/^[ \t]*\*\s*$/m', '', $data));
+                if (trim($data) == '*') {
```

Comment:
please use === here

Reply:
There's some weird stuff going on in there, I just removed the commit for the docblock parser, as it's a time waste.